File: android/app/src/main/java/com/example/attendance/domain/model/Attendance.kt
package com.example.attendance.domain.model

import java.util.Date

enum class AttendanceStatus {
    OFF,        // 勤務外
    WORKING,    // 勤務中
    LEFT        // 退勤済
}

data class Attendance(
    val id: String = "",
    val userId: String = "",
    val date: String = "", // YYYY-MM-DD
    val clockIn: Date? = null,
    val clockOut: Date? = null,
    val status: AttendanceStatus = AttendanceStatus.OFF,
    val workTimeMinutes: Int? = null,
    val createdAt: Date? = null,
    val updatedAt: Date? = null
)

File: web/src/types/Attendance.ts
import { Timestamp } from 'firebase/firestore';

export interface AttendanceRecord {
    id: string;
    userId: string;
    userName?: string; // Enhanced with joined user data if needed
    date: string;
    clockIn: Timestamp | null;
    clockOut: Timestamp | null;
    status: 'working' | 'break' | 'left' | 'absent';
    workTimeMinutes: number;
    updatedAt: Timestamp;
}
